### PR TITLE
Swiss Army Knife and Matches improvements

### LINKED ISSUE #29020: Can't properly use the Swiss Army Knife.
When you hold a crowbar on harm intent, you automatically open firelocks. You can't with the Swiss Army Knife's Crowbar.
You also can't butcher with the Swiss Army Knife's Kitchen Knife/Hatchet/Screwdriver or light oxygen/plasmafires with the Swiss Army Knife's Zippo/strike anywhere match.
[consistency]

### LINKED ISSUE #33051: Swiss Army Knife scroll issues.
#### Description of issue
You can Shift+Scrollwheel to quickly move through the swiss army knife modules, this lets you access the stats of the different tools but sometimes the flags don't update properly. I had the screwdriver out and scrolled to the hatchet and couldn't harvest towercaps, but when I closed and opened the SAK on the hatchet manually it worked fine.
#### Difference between expected and actual behavior
Expected scroll to be identical to close-open then select.
Had to close the SAK and manually select the hatchet.
#### Steps to reproduce
Plant towercaps.
Get swiss army knife.
Select a non-sharp module.
Scroll to hatchet.
Try and fail to cut the towercaps.
Close and open the swiss army knife.
Select the hatchet.
Try and succeed when cutting the towercaps.
[bugfix]

### LINKED ISSUE #34125: Swiss army knife's Zippo lighter and match emit no light
(WEB REPORT BY: tinysaturn REMOTE: 206.221.180.138:7777)
# Revision
d776b3a2fd4a8a847a9f52c9630af610d21c5635

# Description
Unlike normal matches and zippo lighters, the swiss army knife's lighter and match emit no light when they're the active tool.

### PR DESCRIPTION
**Pull request by DeityLink**
Fixes #34125
Fixes #33051
Fixes #29020

:cl:
* rscadd: Running into a Firelock on harm intent while holding a Switchtool with a deployed Crowbar module now opens the Firelock as if you were holding a Crowbar.
* soundadd: Matches and Strike Anywhere Matches now make a sound when you light them up.
* bugfix: Using the Shift and MouseWheel shortcut to swap switchtool modules now properly updates sharpness flags.
* bugfix: The knife on a Swiss Army Knife can now butcher corpses.
* bugfix: The Strike Anywhere Match on a Swiss Army Knife now visibly lights up (and never burns out).
* bugfix: When holding a Swiss Army Knife with an equipped Zippo lighter or lit up Strike Anywhere Match, you now properly emit light, and light up airborne plasma.

### CODE REVIEW

**Comment by Eneocho** [comment]
Closes #34125.
Also, since you're bugfixing the swiss army knives, mind taking a look at these if you can?
#29020 This one's partially fixed, as this does fix the butchering issue. But can you light fuel/plasmafires with the swiss zippo/match? There's also the thing where holding the swiss crowbar while on HARM intent doesn't auto open the firelocks like a regular crowbar does.
#33051 Shift scrollwheeling doesn't properly update the flags, so you can have the hatchet out, but not the sharp edge flag, so you're unable to cut shit.
#32447 Laser scalpels being unable to work as a cautery when inside a med switchtool.
#32828 Holo switchtools can lose tools if you "stick them inside" a cheese wheel or similar, this breaks the holotool, as you can't re-swipe the data disks.
#32805, #18704 and #17584 are pretty much the same as the holo switchtool one.
There's also one small mostly harmless bug where the shift scrollwheel remembers its last selected item over the current one, so if you scrolled to the crowbar (second item) and then selected the pen manually (last item), when scrolling up one, you wouldn't get to the second to last item, but rather the first one.

**Comment by Cartlord** [comment]
just for the record: you can solve the laser-scalpel-in-switchtool thing by using a screwdriver on the scalpel while it's in cautery mode, which seperates it into a laser scalpel and a laser cautery, which can both be added to the switchtool.

**Comment by Eneocho** [comment]
> just for the record: you can solve the laser-scalpel-in-switchtool thing by using a screwdriver on the scalpel while it's in cautery mode, which seperates it into a laser scalpel and a laser cautery, which can both be added to the switchtool.

Indeed, It was mentioned on the linked issue, but it stands to be fixed (Just add both the scalpel AND the cautery when you insert a laser scalpel, and eject them individually if you need em ejected).

**Comment by pedr69** [comment]
>Laser scalpels being unable to work as a cautery when inside a med switchtool.

this is a good thing
imagine opening someone up and separating the liver when suddenly the cautery kicks in and closes the wound

**Comment by DeityLink** [comment]
That'll be it. I'm not dealing with holo-switchtools or mechas in this PR.

**Comment by pedr69** [comment]
possible plastic dupe exploit, i saw a guy with a switchtool crowbarring infinite plastic off a window
when i did it with a normal crowbar it worked as intended

File: __DEFINES/_macros.dm [hunk 1]
```diff
@@ -176,11 +176,11 @@
 
 #define ismultitool(A) istype(A, /obj/item/device/multitool)
 
-#define iscrowbar(A) istype(A, /obj/item/tool/crowbar)
+#define iscrowbar(A) istype(A, /obj/item/tool/crowbar) || (istype(A, /obj/item/weapon/switchtool) && istype(A:deployed, /obj/item/tool/crowbar))
```
Comment (pedr69) [inline __DEFINES/_macros.dm:179]: causes #34308

Comment (pedr69) [reply __DEFINES/_macros.dm:179]: what the hell
this also lets you create one-way windows if you crowbar a fulltile rwindow(also infinite plastic)

Comment (DorainGrey) [reply __DEFINES/_macros.dm:179]: Yes, I saw this too crazy code there to be on base window.

**Pull request merged**